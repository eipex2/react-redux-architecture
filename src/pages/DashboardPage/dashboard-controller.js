//1. create map state to props and return `sessionState`
export const mapStateToProps = ({ sessionState }) => ({
    ...sessionState
})

