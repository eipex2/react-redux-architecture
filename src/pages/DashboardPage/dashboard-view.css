.page-view {
    display:grid;
    text-align:center;
    align-items: center;
    grid-template-rows:1fr;
    grid-template-columns:1fr;
}