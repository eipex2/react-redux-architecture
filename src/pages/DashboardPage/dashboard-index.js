//1. import connect from react-redux
import { connect } from 'react-redux'

//2.import mapStateToProps from our controller
import { mapStateToProps } from './dashboard-controller'

//3. import the view
import DashboardView from './dashboard-view'

//4. connect your view to your redux state
export default connect(mapStateToProps)(DashboardView)