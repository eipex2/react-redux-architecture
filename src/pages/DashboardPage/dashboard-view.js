import React from 'react'

import { Page } from '../../components'

import './dashboard-view.css'

const DashboardView = (props) => {
    return (
        <Page className="page-view">
            <div>
                <h1>Congratulations {props.name}</h1>
                <h3>you are logged in <span role="img" aria-label="smiling-emoji">😁</span></h3>
            </div>
        </Page>
    )
}

export default DashboardView