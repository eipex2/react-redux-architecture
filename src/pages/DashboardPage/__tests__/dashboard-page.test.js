import React from 'react'
import { shallow } from 'enzyme'

import DashboardView from '../dashboard-view'
import DashboardPage from '../dashboard-index'


describe('[<DashboardPage>', () => {

    const mockUser = {
        email: 'foo@bar.com',
        password: '1234'
    }

    const mockStore = {
        getState: jest.fn(() => ({
            sessionState: {
                user: mockUser
            }
        })),
        subscribe: jest.fn(),
        dispatch: jest.fn()
    }

    it('s', () => {
        expect(true).toBe(true)
    })

    it('shd render <DashboardView /> and view shd match snapshot', () => {
        const wrapper = shallow(<DashboardView />)

        expect(wrapper.getElement()).toMatchSnapshot()
    })

    it('shd have session state', () => {
        const wrapper = shallow(<DashboardPage store={mockStore} />)

        expect(wrapper.prop('user')).toEqual(mockUser)
    })
})