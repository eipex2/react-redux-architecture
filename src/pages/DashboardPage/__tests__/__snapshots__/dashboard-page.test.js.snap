// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<DashboardPage> shd render <DashboardView /> and view shd match snapshot 1`] = `
<Page
  className="page-view"
>
  <div>
    <h1>
      Congratulations 
    </h1>
    <h3>
      you are logged in 
      <span
        aria-label="smiling-emoji"
        role="img"
      >
        😁
      </span>
    </h3>
  </div>
</Page>
`;
